File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        

        dic = {}
        
        for i in range(len(nums)):
            
            pair = target-nums[i]
            
            if pair in dic:
                return [ i  , dic[pair]]
            dic[nums[i]] = i 

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        
        map<int, int> hash; 
        vector<int> ret; 
        
        for(int i=0; i <nums.size(); ++i){
            
            int pair = target - nums[i];
            if( hash.find(pair) != hash.end() ){
                ret.push_back(hash[pair] );
                ret.push_back(i);
                
                return ret;
            }
            hash[nums[i]] =i; 
        }
        
        
        
        return ret; 
    }
};